clients/src/test/java/org/apache/kafka/common/utils/ExponentialBackoffTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.utils;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ExponentialBackoffTest {
+    @Test
+    public void testExponentialBackoff() {
+        long scaleFactor = 100;
+        int ratio = 2;
+        long termMax = 2000;
+        double jitter = 0.2;
+        ExponentialBackoff exponentialBackoff = new ExponentialBackoff(
+                scaleFactor, ratio, termMax, jitter
+        );
+
+        for (int i = 0; i <= 100; i++) {
+            for (int n = 0; n <= 4; n++) {
+                assertEquals(scaleFactor * Math.pow(ratio, n), exponentialBackoff.term(n),
+                        scaleFactor * Math.pow(ratio, n) * jitter);
+            }
+            assertTrue(exponentialBackoff.term(1000) <= termMax * (1 + jitter));
```

Comment:
I think that it would be good to test with `n=5` as well to ensure that `termMax` is already applied from it. Perhaps, we could do the following:
```
for (int n = 0; n <= 100; n++) {
  if (n <= 4)
    assertEquals(scaleFactor * Math.pow(ratio, n), exponentialBackoff.term(n),
       scaleFactor * Math.pow(ratio, n) * jitter);
  else
    assertTrue(exponentialBackoff.term(1000) <= termMax * (1 + jitter));
}
```

Reply:
Thanks for the suggestion. I've adopted this in my test.